Sources/NIOHTTP2/HTTP2StreamChannel.swift

```diff
@@ -224,22 +219,46 @@ final class HTTP2StreamChannel: Channel, ChannelCore {
                 self.autoRead = autoRead
                 if let initializer = initializer {
                     initializer(self).whenComplete { result in
-                        switch result {
-                        case .success:
-                            self.postInitializerActivate(promise: promise)
-                        case .failure(let error):
-                            self.configurationFailed(withError: error, promise: promise)
-                        }
+                        self.onInitializationResult(result.map { self }, promise: promise)
                     }
                 } else {
-                    self.postInitializerActivate(promise: promise)
+                    self.postInitializerActivate(promise: promise, output: self)
                 }
             case .failure(let error):
                 self.configurationFailed(withError: error, promise: promise)
             }
         }
     }
 
+    internal func configure<Output>(initializer: @escaping NIOHTTP2Handler.StreamInitializerWithOutput<Output>, userPromise promise: EventLoopPromise<Output>?) {
```

Comment:
We might one to add `@inlinable` to all the generic methods here now so that they get properly specialised. This needs to go up the whole call stack. Might be something for the future though once we drop the SPI.

Reply:
Yeah I think we should do this once the API has settled down and we remove SPI